pkg/hub/controllers/cluster/reconciler.go

```diff
@@ -62,23 +61,46 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		}
 	}
 
-	if cr.Status.ClusterHealth == nil {
-		cr.Status.ClusterHealth = &hubv1alpha1.ClusterHealth{}
-	}
-
-	if err := r.updateClusterHealthStatus(ctx, cr); err != nil {
-		log.Error(err, "unable to update cluster health status")
-	}
-
-	cr.Status.ClusterHealth.LastUpdated = metav1.Now()
-	if err := r.Status().Update(ctx, cr); err != nil {
-		log.Error(err, "unable to update cluster CR")
+	if err := r.updateHealthWithRetry(ctx, cr); err != nil {
 		return reconcile.Result{Requeue: true}, err
 	}
-
+	log.Info("sucessfully updated cluster info on controller")
 	return reconcile.Result{RequeueAfter: ReconcileInterval}, nil
 }
 
+func (r *Reconciler) updateHealthWithRetry(ctx context.Context, cr *hubv1alpha1.Cluster) error {
+	log := logger.FromContext(ctx)
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		err := r.Get(ctx, types.NamespacedName{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		}, cr)
+		if err != nil {
+			log.Error(err, "Error while getting cluster CR retrying")
+			return err
+		}
+		if cr.Status.ClusterHealth == nil {
+			cr.Status.ClusterHealth = &hubv1alpha1.ClusterHealth{}
+		}
+		if err := r.updateClusterHealthStatus(ctx, cr); err != nil {
+			log.Error(err, "unable to update cluster health status")
```

Comment:
return error

Reply:
if err := r.updateClusterHealthStatus(ctx, cr); err != nil {
			log.Error(err, "unable to update cluster health status")
this hsould be inside time's condition